packages/list-exports/index.js

```diff
@@ -543,6 +544,60 @@ function traverseExportsSubdir({
 	}
 }
 
+function mapAndEmitPattern({
+	packageDir, category, tree, lhs, rhs, conditionChain, problems, filteredFiles,
+}) {
+	const lhsStar = lhs.indexOf('*');
+	const rhsStar = rhs.indexOf('*');
+	const lhsPrefix = lhs.slice(0, lhsStar);
+	const lhsSuffix = lhs.slice(lhsStar + 1);
+	const rhsPrefix = rhs.slice(0, rhsStar);
+	const rhsSuffix = rhs.slice(rhsStar + 1);
+
+	const candidates = fg.sync(rhs.replace(/^\.\//, ''), {
+		cwd: packageDir,
+		dot: false,
+		onlyFiles: true,
+		followSymbolicLinks: false,
+	});
+
+	let matchedAny = false;
+	for (const rel of candidates) {
```

Comment:
instead of for-of and a boolean, this can be `return candidates.some((rel) => {` and `return true` in the line that currently sets `matchedAny` to true.